### PR TITLE
ci:  arm64 support

### PR DESCRIPTION
**Pull request by jneo8**
- Fix issue #314, missing testing model on crashdump
- Add arm64 functional test
- Add arm64 charm build

### CODE REVIEW

**Comment by jneo8** [comment]
I can't make grafana-agent works on arm64 machine, [issue open](https://github.com/canonical/grafana-agent-operator/issues/184)

File: tests/functional/bundle.yaml.j2 [hunk 1]
```diff
@@ -7,16 +7,20 @@ series: {{ series }}
...
   ubuntu:
     charm: ubuntu
     num_units: 1
+    constraints: arch={{ arch }}
     to:
       - "0"
   grafana-agent:
     charm: grafana-agent
     channel: stable
+    series: {{ series }}
```
Comment (chanchiwai-ray) [inline tests/functional/bundle.yaml.j2:22]: You have globally define `series`, do you still need it? And why pin the revision?

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -20,70 +20,71 @@ concurrency:
...
+    #   - name: Run unit tests
+    #     run: tox -e unit
 
   func:
-    needs:
-      - lint
-      - unit
+      # needs:
+      #   - lint
+      #   - unit
```
Comment (samuelallan72) [inline .github/workflows/check.yaml:78]: Reminder to uncomment the checks above before merging :)

**Review by samuelallan72 — CHANGES_REQUESTED** [review]
I know the constraints for grafana-agent across architectures in the bundle is still being worked on, but I left some comments on other areas in the PR.  Thanks :)

File: tests/functional/test_charm.py [hunk 3]
```diff
@@ -80,6 +85,7 @@ async def test_build_and_deploy(  # noqa: C901, function is too complex
             "sas2ircu-bin": "empty-resource",
             "sas3ircu-bin": "empty-resource",
         },
+        arch=architecture,
```
Comment (chanchiwai-ray) [inline tests/functional/test_charm.py:88]: Once you have model constraint, may be this is not necessary

Comment (jneo8) [reply tests/functional/test_charm.py:88]: Removed

File: tests/functional/conftest.py [hunk 4]
```diff
@@ -40,6 +41,13 @@ def pytest_addoption(parser):
 def series(request):
     return request.config.getoption("--series")
 
+@pytest.fixture(scope="module")
+def architecture():
+    platform = get_os_platform()
+    if platform.architecture == Architecture.AARCH64:
+        return "arm64"
+    return "amd64"
```
[the remark this answers is not in the document]
Comment (jneo8) [reply tests/functional/conftest.py]: Updated

**Review by chanchiwai-ray — DISMISSED** [review]

File: charmcraft.yaml [hunk 5]
```diff
@@ -14,20 +14,23 @@ bases:
   - build-on:
     - name: ubuntu
       channel: "22.04"
-      architectures: ["amd64"]
+      architectures: ["amd64", "arm64"]
```
Comment (samuelallan72) [inline charmcraft.yaml:17]: Sorry I missed this earlier - the bases need to be split (we shouldn't run a charm built on a different architecture).  See https://github.com/canonical/charm-prometheus-libvirt-exporter/pull/37/files#diff-8af267359854d30b091fde763088e458bf27c8fedc056895f3104ef05b61f4b5 for example

**Review by samuelallan72 — CHANGES_REQUESTED** [review]

File: charmcraft.yaml (hunk 5 above)
Comment (jneo8) [reply charmcraft.yaml]: Updated

**Review by samuelallan72 — APPROVED** [review]
Thanks, looks good to me :D

**Review by chanchiwai-ray — APPROVED** [review]

**Pull request merged**